tgui/packages/tgui/interfaces/PowerMonitor/Apc.tsx

```diff
@@ -0,0 +1,170 @@
+    <Box inline>
+      On{' '}
+      <Box inline color="grey">
+        (Auto)
+      </Box>
+    </Box>
+  ),
+};
+
+const apcCellState = {
+  [0]: 'Discharging',
+  [1]: 'Charging',
+  [2]: 'Charged',
+};
+
+export const PowerMonitorApcGlobal = (_props, context) => {
+  const { data } = useBackend<PowerMonitorApcData>(context);
+
+  const availableHistory = data.history.map((v) => v[0]);
+  const availableHistoryData = availableHistory.map((v, i) => [i, v]);
+
+  const loadHistory = data.history.map((v) => v[1]);
+  const loadHistoryData = loadHistory.map((v, i) => [i, v]);
+
+  const max = Math.max(...availableHistory, ...loadHistory);
+
+  return (
+    <Stack fill>
+      <Stack.Item width="50%">
+        <LabeledList>
+          <LabeledList.Item label="Total Power">{formatPower(data.available)}</LabeledList.Item>
+        </LabeledList>
+        <Chart.Line
+          mt="5px"
+          height="5em"
+          data={availableHistoryData}
+          rangeX={[0, availableHistoryData.length - 1]}
+          rangeY={[0, max]}
+          strokeColor="rgba(1, 184, 170, 1)"
+          fillColor="rgba(1, 184, 170, 0.25)"
+        />
+      </Stack.Item>
+      <Stack.Item width="50%">
+        <LabeledList>
+          <LabeledList.Item label="Total Load">{formatPower(data.load)}</LabeledList.Item>
+        </LabeledList>
+        <Chart.Line
+          mt="5px"
+          height="5em"
+          data={loadHistoryData}
+          rangeX={[0, loadHistoryData.length - 1]}
+          rangeY={[0, max]}
+          strokeColor="rgba(1, 184, 170, 1)"
+          fillColor="rgba(1, 184, 170, 0.25)"
+        />
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+export const PowerMonitorApcTableHeader = (_props, context) => {
+  return (
+    <>
+      <Table.Cell header>Area</Table.Cell>
+      <Table.Cell header collapsing position="relative">
+        Eqp.
+        <Tooltip content="Equipment" />
+      </Table.Cell>
+      <Table.Cell header collapsing position="relative">
+        Lgt.
+        <Tooltip content="Lighting" />
+      </Table.Cell>
+      <Table.Cell header collapsing position="relative">
+        Env.
+        <Tooltip content="Environment" />
+      </Table.Cell>
+      <Table.Cell textAlign="right" header>
+        Load
+      </Table.Cell>
+      <Table.Cell textAlign="right" header>
+        Cell Charge
+      </Table.Cell>
+      <Table.Cell header>Cell State</Table.Cell>
+    </>
+  );
+};
+
+export const PowerMonitorApcTableRows = (_props, context) => {
+  const { data } = useBackend<PowerMonitorApcData>(context);
+
+  return (
+    <>
+      {data.apcs.map((apc) => (
+        <PowerMonitorApcTableRow key={apc[0]} apc={apc} />
+      ))}
+    </>
+  );
+};
+
+type PowerMonitorApcTableRowProps = {
+  apc: PowerMonitorApcItemData;
+};
+
+const PowerMonitorApcTableRow = (props: PowerMonitorApcTableRowProps, context) => {
+  const { apc } = props;
+  const [ref, equipment, lighting, environment, load, cellCharge, cellCharging] = apc;
```

Comment:
Comment on reason for index-based data vs. key/value data (size of data with a large number of rows).